From a2f9abd2f518a422e0a1347304dda963e9b9c272 Mon Sep 17 00:00:00 2001
From: Alexey Brodkin <alexey.brodkin@gmail.com>
Date: Wed, 5 Jan 2022 16:22:53 +0300
Subject: [PATCH] build infrastructure: silent and verbose modes

Inspired by https://github.com/baresip/baresip/pull/87
to make it easier to see what's being done really during
build, if make run with "V=1".

This is especially important since we do some tricky
run-time checks for some components and the only way to find-out
what was detected and what not is to see full set of definitions
passed in CFLAGS.

Upstream-Status: Submitted [https://github.com/baresip/re/pull/194]

Signed-off-by: Alexey Brodkin <alexey.brodkin@gmail.com>
---
 Makefile | 41 +++++++++++++++++++++++------------------
 1 file changed, 23 insertions(+), 18 deletions(-)

diff --git a/Makefile b/Makefile
index d7e0d60..c63d72c 100644
--- a/Makefile
+++ b/Makefile
@@ -17,6 +17,11 @@ ABI_AGE   := 0
 
 ABI_MAJOR := $(shell expr $(ABI_CUR) - $(ABI_AGE))
 
+# Verbose and silent build modes
+ifeq ($(V),)
+HIDE=@
+endif
+
 PROJECT   := re
 VERSION   := $(VER_MAJOR).$(VER_MINOR).$(VER_PATCH)
 
@@ -79,14 +84,14 @@ all: $(SHARED) $(STATIC)
 
 $(SHARED): $(OBJS)
 	@echo "  LD      $@"
-	@$(LD) $(LFLAGS) $(SH_LFLAGS) $^ $(LIBS) -o $@
+	$(HIDE)$(LD) $(LFLAGS) $(SH_LFLAGS) $^ $(LIBS) -o $@
 
 
 $(STATIC): $(OBJS)
 	@echo "  AR      $@"
-	@$(AR) $(AFLAGS) $@ $^
+	$(HIDE)$(AR) $(AFLAGS) $@ $^
 ifneq ($(RANLIB),)
-	@$(RANLIB) $@
+	$(HIDE)$(RANLIB) $@
 endif
 
 libre.pc:
@@ -105,21 +110,21 @@ libre.pc:
 
 $(BUILD)/%.o: src/%.c $(BUILD) Makefile $(MK) $(MODMKS)
 	@echo "  CC      $@"
-	@$(CC) $(CFLAGS) -c $< -o $@ $(DFLAGS)
+	$(HIDE)$(CC) $(CFLAGS) -c $< -o $@ $(DFLAGS)
 
 
 $(BUILD): Makefile $(MK) $(MODMKS)
-	@mkdir -p $(patsubst %,$(BUILD)/%,$(sort $(dir $(SRCS))))
-	@touch $@
+	$(HIDE)mkdir -p $(patsubst %,$(BUILD)/%,$(sort $(dir $(SRCS))))
+	$(HIDE)touch $@
 
 
 .PHONY: clean
 clean:
-	@rm -rf $(SHARED) $(STATIC) libre.pc test.d test.o test $(BUILD)
+	$(HIDE)rm -rf $(SHARED) $(STATIC) libre.pc test.d test.o test $(BUILD)
 
 
 install: $(SHARED) $(STATIC) libre.pc
-	@mkdir -p $(DESTDIR)$(LIBDIR) $(DESTDIR)$(LIBDIR)/pkgconfig \
+	$(HIDE)mkdir -p $(DESTDIR)$(LIBDIR) $(DESTDIR)$(LIBDIR)/pkgconfig \
 		$(DESTDIR)$(INCDIR) $(DESTDIR)$(MKDIR)
 	$(INSTALL) -m 0644 $(shell find include -name "*.h") \
 		$(DESTDIR)$(INCDIR)
@@ -135,24 +140,24 @@ endif
 	$(INSTALL) -m 0644 $(MK) $(DESTDIR)$(MKDIR)
 
 uninstall:
-	@rm -rf $(DESTDIR)$(INCDIR)
-	@rm -rf $(DESTDIR)$(MKDIR)
-	@rm -f $(DESTDIR)$(LIBDIR)/$(SHARED)
-	@rm -f $(DESTDIR)$(LIBDIR)/$(SHARED_SONAME)
-	@rm -f $(DESTDIR)$(LIBDIR)/$(STATIC)
-	@rm -f $(DESTDIR)$(LIBDIR)/pkgconfig/libre.pc
+	$(HIDE)rm -rf $(DESTDIR)$(INCDIR)
+	$(HIDE)rm -rf $(DESTDIR)$(MKDIR)
+	$(HIDE)rm -f $(DESTDIR)$(LIBDIR)/$(SHARED)
+	$(HIDE)rm -f $(DESTDIR)$(LIBDIR)/$(SHARED_SONAME)
+	$(HIDE)rm -f $(DESTDIR)$(LIBDIR)/$(STATIC)
+	$(HIDE)rm -f $(DESTDIR)$(LIBDIR)/pkgconfig/libre.pc
 
 -include test.d
 
 test.o:	test.c Makefile $(MK)
 	@echo "  CC      $@"
-	@$(CC) $(CFLAGS) -c $< -o $@ $(DFLAGS)
+	$(HIDE)$(CC) $(CFLAGS) -c $< -o $@ $(DFLAGS)
 
 test$(BIN_SUFFIX): test.o $(SHARED) $(STATIC)
 	@echo "  LD      $@"
-	@$(LD) $(LFLAGS) $< -L. -lre $(LIBS) -o $@
+	$(HIDE)$(LD) $(LFLAGS) $< -L. -lre $(LIBS) -o $@
 
 sym:	$(SHARED)
-	@nm $(SHARED) | grep " U " | perl -pe 's/\s*U\s+(.*)/$${1}/' \
+	$(HIDE)nm $(SHARED) | grep " U " | perl -pe 's/\s*U\s+(.*)/$${1}/' \
 		> docs/symbols.txt
-	@echo "$(SHARED) is using `cat docs/symbols.txt | wc -l ` symbols"
+	$(HIDE)echo "$(SHARED) is using `cat docs/symbols.txt | wc -l ` symbols"
-- 
2.25.1

